src/java/org/apache/cassandra/db/DeletionTime.java

```diff
@@ -195,42 +195,100 @@ public static Serializer getSerializer(Version version)
-            return mfda == Long.MIN_VALUE && localDeletionTimeUnsignedInteger == Cell.NO_DELETION_TIME_UNSIGNED_INTEGER
-                 ? LIVE
-                 : new DeletionTime(mfda, localDeletionTimeUnsignedInteger);
+            int flags = in.readByte();
+            if ((flags & IS_LIVE_DELETION) != 0)
+                return LIVE;
+            else
+            {
+                // Read the 7 bytes
+                int bytes4 = in.readInt();
+                int bytes2 = in.readShort();
+                int bytes1 = in.readByte();
+
+                long mfda = sevenBytesToMFDA(bytes4, bytes2, bytes1);
+                int localDeletionTimeUnsignedInteger = in.readInt();
+
+                return new DeletionTime(mfda, localDeletionTimeUnsignedInteger);
+            }
         }
 
         public DeletionTime deserialize(ByteBuffer buf, int offset)
         {
-            int localDeletionTimeUnsignedInteger = buf.getInt(offset);
-            long mfda = buf.getLong(offset + 4);
-            return mfda == Long.MIN_VALUE && localDeletionTimeUnsignedInteger == Cell.NO_DELETION_TIME_UNSIGNED_INTEGER
-                   ? LIVE
-                   : new DeletionTime(mfda, localDeletionTimeUnsignedInteger);
+            int flags = buf.get(offset);
+            if ((flags & IS_LIVE_DELETION) != 0)
+                return LIVE;
+            else
+            {
+                // Read the 7 bytes
+                int bytes4 = buf.getInt(offset + 1);
+                int bytes2 = buf.getShort(offset + 5);
+                int bytes1 = buf.get(offset + 7);
+                
+                long mfda = sevenBytesToMFDA(bytes4, bytes2, bytes1);
+                int localDeletionTimeUnsignedInteger = buf.getInt(offset + 8);
+
+                return new DeletionTime(mfda, localDeletionTimeUnsignedInteger);
+            }
         }
 
         public void skip(DataInputPlus in) throws IOException
         {
-            in.skipBytesFully(4 + 8);
+            int flags = in.readByte();
+            if ((flags & IS_LIVE_DELETION) != 0)
+                // We read the flags already and there's nothing left to skip over
+                return;
+            else
+                // We read the flags already but there is mfda and ldt to skip over still
+                in.skipBytesFully(TypeSizes.LONG_SIZE - 1 + TypeSizes.INT_SIZE);
         }
 
         public long serializedSize(DeletionTime delTime)
         {
-            return TypeSizes.sizeof(Integer.MAX_VALUE)
-                   + TypeSizes.sizeof(delTime.markedForDeleteAt());
+            if (delTime.equals(LIVE))
+                // Just the flags
+                return 1;
+            else
+                // 1 for the flags, 7 for mfda and 4 for ldt
+                return TypeSizes.INT_SIZE
+                       + TypeSizes.LONG_SIZE;
+        }
+
+        private long sevenBytesToMFDA(int bytes4, int bytes2, int bytes1)
+        {
+                long mfda = bytes4 & 0xFFFFFFFFL;
```

Comment:
it's only 4 bytes here, but still `(1L << 32) - 1` might be preferable

Reply:
Do you mind I leave it as it is to be the same format as the as the next 2 lines? being only 4 bytes it's no eye offensive as the `MAX_MFDA` one

Reply:
nit: either 0xff.. or 0xFF..

Reply:
Right you are

Reply:
Fixed in review commits. Resolving to clear the PR a bit but feel free to reopen if you disagree.